### PR TITLE
Magic balance pass/spell list changes

### PR DESCRIPTION
**Pull request by RadiantFlash7**
## About The Pull Request
This PR Re-balances a lot of spells in the mage spell list. Currently the most common ones people ever take is fireball, lightning bolt, and repulse for combat reasons. Some of the other ones such as Arcane bolt, spitfire were some of the most useless things I've ever seen, and either needed improvement or spellpoint decreases. This PR aims to rebalance the spell list for things to be worth their spell cost, or reduced in cost.
From the top:

**- Fireball:** Reduced in cost from 4 to 3. Fireball used to be the best spell bar none, because of the way firestacks did damage. With how firestacks were nerfed, unless someone is stupid enough to stand on the fire after being hit by the fireball, you get anywhere from 20 to 60ish damage or so on use.

**-Lightning bolt:** Will now shock everyone equally. Apparently, gloves used to completely negate this from doing anything, ever. Gloves have lost their anti-lightning magic

**-Spitfire:** Reworked. Formerly a worthless damage spell with 'fast' cool down, It's now a burst fired attack. Upon activation, it will fire a burst of three fire bolts that deal 15 burn damage each, for a total of 45 damage, assuming all hit- a fairly big if. This spell has minor homing features, but in likely hood you won't notice unless you're running away from it, given the speed of the projectiles. Still fairly easy to dodge, however if moving perpendicular to caster.

**-Arcane bolt:** Bumped up to 45 brute damage, and cost increased to 3. This spell is the equivalent of shooting an arrow at someone if you're decent at using a bow and arrow. Does have slight armor piercing.

**-Forcewall:** Formerly a noob trap spell, This was more or less only used by people to trap themselves in with someone else. It has been changed to allow the caster to walk through it, at the cost of reducing the wall's existance to 10 seconds instead of 20.

**-Ensare:** A hold spell that was effectively worthless due to the short time it held people in place. It was not long enough to cast even the shortest castable spell on our spell list, leaving no way to take advantage of holding someone, aside from fleeing. immobalize Duration buffed to 4 seconds from 2.5

**-Bladeburst:** This spell was probably the hardest one to figure out what to do with. It's atrocious. Extremely difficult to land a hit with, and the damage it did spread all over the body, making it effectively worthless in causing any significant harm on someone. I reworked this spell to instead direct all that damage to a randomly picked leg, and fracture a leg bone. This might be an over compensation, but it's hard to tell, and as I said, it's extremely hard to even land a hit with this. Might need a point increase to 3, we'll have to see.

**-Haste:** A interesting spell, It's duration was increased to 1.5 minutes, and cooldown of casting it halfed to 2.5 minutes from 5. It's cost was reduced to 2 spell points from 3, and it recieved near double the stamina to cast. Better now for use to buff others, though you can still buff yourself mid fight in a pinch.

image
image
image

----

This PR also adds a new spell that I couldn't resist adding- the traditional find familiar spell. Spawns a spectral looking volf that is hostile to everyone except the caster, and disappears after 30 seconds.

---
Finally, This PR adjusts arcane spell XP gain. In normal gameplay as a mage, you would /never/ level up, even as a novice, unless you intentionally grinded spellcasting at a wall or trees, or-anything. As an example for how bad this was, It would take /50/ fire balls to advance from apprentice, to journeyman. Further more due to the way XP was calculated based on fatigue spent, the higher INT level you were as a mage, the less fatigue you used, meaning the better mage you were, the worse you were at learning. Thus:
The overall spell xp multiplier has been increased from .3 to .5, and a multiplier has been added based on the mage's INT level, divided by ten. This functions the same way a archers bow damage is increased.
EX: an 11 int mage will gain 1.1 times the magic xp they would have. Likewise, a 15INT mage gains 1.5 xps the xp.

## Why It's Good For The Game
Currently the most common ones people ever take is fireball, lightning bolt, and repulse for combat reasons. Some of the other ones such as Arcane bolt, spitfire were some of the most useless things I've ever seen, and either needed improvement or spell point decreases. It was very clear that some spells were just horrendiously bad with no redeeming qualities. Now, all the spells should be relatively more inline with each other, letting people feel like they have more of a choice with what to pick and use.

Also, No one should have to spam cast 50 spells to level up from apprentice to journeyman. It gets much worse the higher the level, so the xp gain had to be increased.

I'm saving most of the other spells I intend on adding at some point for another PR: This one is predominantly for rebalancing existing ones.

### CODE REVIEW

**Comment by Nakalon** [comment]
Wheres test proof 😡

**Comment by Nakalon** [comment]
The sprite is absolutely stunning by the way

**Comment by RadiantFlash7** [comment]
> Wheres test proof 😡

I mean. The PR is mooostly balance number crunching. I tested it in a test server ofc, but theres not much for me to really showcase. I can throw up a video or so, maybe.

**Comment by RadiantFlash7** [comment]
> The sprite is absolutely stunning by the way

It's just a color-swapped volf with alpha set to 155, but I appreciate it!

**Comment by Nakalon** [comment]
> > Wheres test proof 😡
> 
> I mean. The PR is mooostly balance number crunching. I tested it in a test server ofc, but theres not much for me to really showcase. I can throw up a video or so, maybe.

what im saying, testing proof for the new skill you just added silly, also refer to my REVIEWWWWWWWWW

**Review by Nakalon — CHANGES_REQUESTED** [review]

**Comment by RadiantFlash7** [comment]
> > > Wheres test proof 😡
> > 
> > 
> > I mean. The PR is mooostly balance number crunching. I tested it in a test server ofc, but theres not much for me to really showcase. I can throw up a video or so, maybe.
> 
> what im saying, testing proof for the new skill you just added silly, also refer to my REVIEWWWWWWWWW

I was about to say, theres no review listed!!! But there we go....

File: code/modules/mob/living/simple_animal/rogue/creacher/familiar.dm [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/mob/living/simple_animal/hostile/retaliate/rogue/volf/familiar
```
[the remark this answers is not in the document]
Comment (RadiantFlash7) [reply code/modules/mob/living/simple_animal/rogue/creacher/familiar.dm:1]: that is- a very silly typo I made, in fact.

**Comment by Nakalon** [comment]
> > > > Wheres test proof 😡
> > > 
> > > 
> > > I mean. The PR is mooostly balance number crunching. I tested it in a test server ofc, but theres not much for me to really showcase. I can throw up a video or so, maybe.
> > 
> > 
> > what im saying, testing proof for the new skill you just added silly, also refer to my REVIEWWWWWWWWW
> 
> I was about to say, theres no review listed!!! But there we go....

I didn't know how to review...

File: code/modules/mob/living/simple_animal/rogue/creacher/familiar.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nakalon) [reply code/modules/mob/living/simple_animal/rogue/creacher/familiar.dm:1]: > that is- a very silly typo I made, in fact.

That's normal, but do clear up the duplicated vars and the code will be much more organized and readable. Cheers!!

**Comment by Nakalon** [comment]
I like how you added more info to the PR and test evidence! Good work!

**Comment by RadiantFlash7** [comment]
> fucking balance pass's.... motivation always sucked right out of me on merges when I pass one like this I gotta magnifying glass..... will get around to it after discussion...

Feel free to ask whatever questions about whatever you like, I'm happy ro answer

Currently theres only one thing I haven't fussed with that I wanted to for spells and that was lightning bolt. It seems very inconsistent whether someone can be electrocuted, and I'm not sure whats preventing it or not.

I've tried shocking people in full plate on the live server, only for it to fail, yet doing the same on my test server they got electrocuted.

**Comment by DatFrostys** [comment]
> I've tried shocking people in full plate on the live server, only for it to fail, yet doing the same on my test server they got electrocuted.

I believe gloves protects you from being electrocuted

**Comment by RadiantFlash7** [comment]
> > I've tried shocking people in full plate on the live server, only for it to fail, yet doing the same on my test server they got electrocuted.
> 
> I believe gloves protects you from being electrocuted

You're correct and this is cancer. How did no one on roguetown catch this? 💢 
Fixed now.

**Comment by RadiantFlash7** [comment]
So far no ones really taken any issues with this. I'd say it's probably ready to be test merged.

**Comment by RadiantFlash7** [comment]
> I put a test merge on this... magic is a slippery slope I watch hearth drown from it...

Well, I don't quite think making the other spells worth taking is going down that slope quite yet. I thought I did fairly okay on balancing it all.

**Comment by radarseas** [comment]
honestly, I'd keep forcewall at 16 seconds, its GOOD against people with crossbows and bows or sprinting chasers, but anyone with melee is going to just smash through it in seconds anyway.

**Comment by RadiantFlash7** [comment]
> honestly, I'd keep forcewall at 16 seconds, its GOOD against people with crossbows and bows or sprinting chasers, but anyone with melee is going to just smash through it in seconds anyway.

We'll give it a test merge at 10, and see. Being able to walk through it is fairly useful, and you can get off 2 spells if you're quick.

**Comment by HissyNays** [comment]
Haste requiring more stamina to cast is bad as it's best used on self to evade danger in critical moment but you might be wasting so much stamina that you were better off just sprinting. It's a really tame spell that is still regardless of anything and in practice best used on self. Otherwise it desperately needed duration and cooldown improvements, the cost change is also something I can't say is bad.

Spitfire getting homing ability is strange but I understand why, hitting a burst of projectiles is extremely hard. I think instead this should be trait for magic missile, since clasically it does have that.
For spitfire just make it single fireball, decrease it's cooldown and stamina usage so it's more spammable. Spitfire also has huge problem, it cannot concentrate damage on single limb, all damage spreads all over body making significantly less useful. Additionally the blind should happen only on tile it hits, it's biggest benefit is that tiny blind but it often blinds you aswell making it a double edged sword. Otherwise fireball is still flat out better option so spitfire needs tweaking of price and power both or just power.

Magic missile (arcane bolt) cost increase is bad, infact i'm really dissapoitned about it :(. It's best trait was that it was a cheap spammable damage spell and I think it's best be kept that way. Cost increase despite other improvements really killed it.
All it really could use was stab wounds and homing, otherwise no other changes were necessary. And again, fireball is flat out better option for the same price.

LIGHTNING. May god have mercy on us all, you have created the most overpowered spell with this tiny change, you really might want to reverse this. There was an extremely good reason for gloves to be protection against it and it served to make gloves much more desireable. What you should have done is just decrease it's cost to 2 spellpoints to account for it's extreme nicheness. Right now it's equal to greater fireball if not stronger.

Fireball cost decrease is good but that cost decrease actually made it return to being the best spell out of all options.

Forcewall improvements are very good but if you really want to make it good you have to make the wall spawn not infront of caster but inside the caster. Reducing it's life time is also questionable and mostly pointless as any semi-smart opponent will just break it in 3 hits.

Ensnare definitely felt very weak, not sure if that buff will save it still though. It serves as a setup for other spells but it's too pricy when you can just aim right instead.

Blade burst was really not bad but it really lacked ability to inflict wounds which you fixed. It's a great utility spell and price increase will absolutely kill it. Otherwise it was the only multi-z level spell and it was good for that. Before this change I tested it and it seemed to already focus damage on one or the other leg. So not sure why our experiences were different.

Find fammiliar seems like a really good addition, good work. But perhaps there could be an additional spell you receive that basicly just works to command the pet? Click a guy, it targets the guy, click your self, it runs to you, click on ground and it ignores everything untill it gets to the place you commanded it to move. But thats probably really hard to code so it's probably wishful thinking.

In general I think current spellpoint system doesn't work well because spell costs are too rigid, I think amount of spell points you gain should be doubled and prices adjusted accordingly. 
Because 3 points for fireball is too little, 4 is too much and correct price for fireball would be between the two numbers but we can't do that.

**Comment by RadiantFlash7** [comment]
> Haste requiring more stamina to cast is bad as it's best used on self to evade danger in critical moment but you might be wasting so much stamina that you were better off just sprinting. It's a really tame spell that is still regardless of anything and in practice best used on self. Otherwise it desperately needed duration and cooldown improvements, the cost change is also something I can't say is bad.

Haste has a decent use in combat in buffing your speed, which is used both to speed yourself up for movement, and to increase your attackspeed/defense for weapons. I would argue it should never really have been used as a 'getaway card' and instead casted at the start of the fight. Hence the increased time durection, and reduced cooldown.
> 
> Spitfire getting homing ability is strange but I understand why, hitting a burst of projectiles is extremely hard. I think instead this should be trait for magic missile, since clasically it does have that. For spitfire just make it single fireball, decrease it's cooldown and stamina usage so it's more spammable. Spitfire also has huge problem, it cannot concentrate damage on single limb, all damage spreads all over body making significantly less useful. Additionally the blind should happen only on tile it hits, it's biggest benefit is that tiny blind but it often blinds you aswell making it a double edged sword. Otherwise fireball is still flat out better option so spitfire needs tweaking of price and power both or just power.

I considered a few options. with spitfire. 45 damage is actually quite a lot, considering no armor actually blocks fire damage. It won't crit someone, or cause bleeding injuries that kill, but it's still quite brutal. Fire damage especially causes a lot more pain then cuts, so if you get hit by the full bursts, you'll go down from pain crit after a couple vollies.
Fireball itself isn't great, in that it's massive damage is  dependant on your subject standing in the fire due to the nerfs to the firestacks. In testing, I found the spitfire to do comparable damage to someone who's moving about, that a fireball would do. (I've also fireballed a person 8 times and they never hit pain crit, because they never stood still in the fire to gather increased firestacks.))
> 
> Magic missile (arcane bolt) cost increase is bad, infact i'm really dissapoitned about it :(. It's best trait was that it was a cheap spammable damage spell and I think it's best be kept that way. Cost increase despite other improvements really killed it. All it really could use was stab wounds and homing, otherwise no other changes were necessary. And again, fireball is flat out better option for the same price.

Magic missile prior was effectively worthless. Keep in mind, as a mage, you need to keep moving. If you're casting spells, you can not block, or dodge at the same time. This is very important to keep in mind as to why mages are also designed as glass cannons. The magic missile projectile speed was increased further( not currently live on the server) in that, unless someones a speedy rogue, you should almost never miss. Which is not the same for fireball or spitfire, significantly slower projectiles. Consider this a much more reliable way of fixing it.
> 
> LIGHTNING. May god have mercy on us all, you have created the most overpowered spell with this tiny change, you really might want to reverse this. There was an extremely good reason for gloves to be protection against it and it served to make gloves much more desireable. What you should have done is just decrease it's cost to 2 spellpoints to account for it's extreme nicheness. Right now it's equal to greater fireball if not stronger.

This spell is mostly good for it's knock down. Gloves blocking it was an /oversight/, not an intentional protection. A spell should never be extremely inconsistent in performance like it was. Consider it similar to repulse, except it only has a single target. Making it consistent is important, IMO.
> 
> Fireball cost decrease is good but that cost decrease actually made it return to being the best spell out of all options.

I see you concern on this one, but I firmly disagree on it being the best choice. Maybe so in pve, where mobs will stand still in a pit of fire, but in comparison for pvp, it's hard to point at fireball and go, 'it's better then arcane bolt and spitfire' as it is now. In fact, I'd argue in pvp, it does less damage then both of those, routinely.

> 
> Forcewall improvements are very good but if you really want to make it good you have to make the wall spawn not infront of caster but inside the caster. Reducing it's life time is also questionable and mostly pointless as any semi-smart opponent will just break it in 3 hits.
The lesser time change is useful for people fighting mages from range. As for where it spawns, Thats a fair point, but I don't really like spawning it directly ontop of the caster. Doesn't fit super well, and looks awkward.
> 
> Ensnare definitely felt very weak, not sure if that buff will save it still though. It serves as a setup for other spells but it's too pricy when you can just aim right instead.

You can now combo ensare with other spells, and perhaps more importantly, it serves as a great support spell if you work in conjunction with a second person.
> 
> Blade burst was really not bad but it really lacked ability to inflict wounds which you fixed. It's a great utility spell and price increase will absolutely kill it. Otherwise it was the only multi-z level spell and it was good for that. Before this change I tested it and it seemed to already focus damage on one or the other leg. So not sure why our experiences were different.

I tested blade burst extensively, and it would do damage all over the body in my testing. I still worry that it may be too strong with breaking a bone, but as I said prior, it's extremely hard to land on another player.

> 
> Find fammiliar seems like a really good addition, good work. But perhaps there could be an additional spell you receive that basicly just works to command the pet? Click a guy, it targets the guy, click your self, it runs to you, click on ground and it ignores everything untill it gets to the place you commanded it to move. But thats probably really hard to code so it's probably wishful thinking.
Given the way mob ai works, it attacks the clost thing first. And given it's a temporary pet that lasts 30 seconds, coding in commands for it wasn't worth the effort.
> 
> In general I think current spellpoint system doesn't work well because spell costs are too rigid, I think amount of spell points you gain should be doubled and prices adjusted accordingly. Because 3 points for fireball is too little, 4 is too much and correct price for fireball would be between the two numbers but we can't do that.
I don't hate the spell point system. it's a bit hard to balance well, but I think keeping damaging spells at 3 and utility/buff spells at 2 is a fairly good benchline.

**Comment by HissyNays** [comment]
Some extra feedback on untouched spells that I think could use buff.

Fetch is extremely niche but it's really satisfying to pull people out from danger. It's effects can be resisted by just running in opposite direction of it. Perhaps it's cost could be lowered to 1 instead of 2?
Or it could be made impossible to negate it just by moving?

Predigistation - I believe it's too expensive, yeah you heard me. Maybe it should just be free starting spell? In the long run it's always waste of spell points and it's biggest benefit can be replaced by a cheap lamptern. 

Featherfall is a lovely use for that one last spell point. It can potentially save your life but it would feel much more right if it saved you from fall damage from 2 z-levels and not just one.

**Comment by RadiantFlash7** [comment]
> Some extra feedback on untouched spells that I think could use buff.
> 
> Fetch is extremely niche but it's really satisfying to pull people out from danger. It's effects can be resisted by just running in opposite direction of it. Perhaps it's cost could be lowered to 1 instead of 2? Or it could be made impossible to negate it just by moving?

Hm. I don't really have a giant issue with lowering fetch cost, tbh. pulling people towards yourself in combat tends to be a bad idea in most situations, and this doesn't really have a whole lot of uses.
> 
> Predigistation - I believe it's too expensive, yeah you heard me. Maybe it should just be free starting spell? In the long run it's always waste of spell points and it's biggest benefit can be replaced by a cheap lamptern.

The free lamptern is fairly useful, if 'm honest. i'm hesitant to just give it out for free off the bar. Especially since it has other uses, albiet small ones.
> 
> Featherfall is a lovely use for that one last spell point. It can potentially save your life but it would feel much more right if it saved you from fall damage from 2 z-levels and not just one.

to be honest, I've never actually touched featherfall. Both in testing, and as a player. I've never had a situation where my magic user needed to jump down a story and not be harmed, where I couldn't just climb down.

**Comment by Rayforx** [comment]
I think it would be fine to make prestidigitation into a free spell, that everyone owns. The original idea for the spells was to make sure that it was free, but it was left on 1 because of the spell increase when it was added. With further testing, it would be fine to change it back to free again. All of the powers it gives are purely roleplay or with minimal use (the lamp, which has a very short radius does a bit of use, but it's a shorter range than a touch)

**Pull request merged**